powerapi-core/src/main/scala/org/powerapi/core/power/Power.scala

```diff
@@ -23,132 +23,163 @@
-  def apply(value: Double, unit: String)    = new RawPower(value, PowerUnitSystem(unit))
+  def lt(other: Power) = this < other
+
+  def lteq(other: Power) = this <= other
+
+  def minus(other: Power) = this - other
+
+  def mul(factor: Double) = this * factor
+
+  def plus(other: Power) = this + other
+}
+
+object RawPower {
 
   // limit on abs. value of powers in their units
   private final val max_mw = Double.MaxValue
   private final val max_w = max_mw / 1000.0
   private final val max_kw = max_w / 1000.0
   private final val max_Mw = max_kw / 1000.0
+
+  def apply(value: Double, unit: PowerUnit) = new RawPower(value, unit)
+
+  def apply(value: Double, unit: String) = new RawPower(value, PowerUnitSystem(unit))
+
+  implicit object RawPowerIsOrdered extends Ordering[RawPower] {
+    def compare(a: RawPower, b: RawPower) = a compare b
+  }
+
 }
 
 /**
- * Defines a power value.
- *
- * @author Loïc Huertas <l.huertas.pro@gmail.com>
- * @author Romain Rouvoy <romain.rouvoy@univ-lille1.fr>
- */
+  * Defines a power value.
+  *
+  * @author Loïc Huertas <l.huertas.pro@gmail.com>
+  * @author Romain Rouvoy <romain.rouvoy@univ-lille1.fr>
+  */
 final class RawPower(val value: Double, val unit: PowerUnit) extends Power {
   private val log = LogManager.getLogger
-  
-  private[this] def bounded(max: Double) = 0.0 <= value && value <= max
-  
+
+  def toWatts = unit.toWatts(value)
+
   require(unit match {
-      case MILLIWATTS => bounded(max_mw)
-      case WATTS      => bounded(max_w)
-      case KILOWATTS  => bounded(max_kw)
-      case MEGAWATTS  => bounded(max_Mw)
-      case _ =>
-        val v = MEGAWATTS.convert(value, unit)
-        0.0 <= v && v <= max_Mw
-    }, "Power value is limited to 1.79e308 mW and cannot be negative")
+    case MILLIWATTS => bounded(max_mw)
+    case WATTS => bounded(max_w)
+    case KILOWATTS => bounded(max_kw)
+    case MEGAWATTS => bounded(max_Mw)
+    case _ =>
+      val v = MEGAWATTS.convert(value, unit)
+      0.0 <= v && v <= max_Mw
+  }, "Power value is limited to 1.79e308 mW and cannot be negative")
+
+  def toKiloWatts = unit.toKiloWatts(value)
+
+  def toMegaWatts = unit.toMegaWatts(value)
 
-  def toMilliWatts = unit.toMilliWatts(value)
-  def toWatts      = unit.toWatts(value)
-  def toKiloWatts  = unit.toKiloWatts(value)
-  def toMegaWatts  = unit.toMegaWatts(value)
   def toUnit(u: PowerUnit) = toMilliWatts / MILLIWATTS.convert(1, u)
-  
+
   override def toString() = s"$value $unit"
-  
+
   def compare(other: Power) = toMilliWatts compare other.toMilliWatts
-  
-  private[this] def safeAdd(a: Double, b: Double): Double = {
-    if ((b > 0.0) && (a > Double.MaxValue - b)) throw new IllegalArgumentException("double overflow")
-    if ((b < 0.0) && (a < -b)) throw new IllegalArgumentException("negative power cannot exists")
-    a + b
-  }
+
+  def toMilliWatts = unit.toMilliWatts(value)
+
+  def +(other: Power) = add(other.value, other.unit)
```

Comment:
[Scalastyle] WARNING: Method name does not match the regular expression '^[a-z][A-Za-z0-9]*(_=)?$'